File: ipc/loesung/02/pipe_writer.py
#!/usr/bin/env python3

import os

if __name__ == '__main__':

    fifo_path = '/tmp/my_pipe'

    # Named Pipe anlegen (einmalig)
    if not os.path.exists(fifo_path):
        os.mkfifo(fifo_path)

    with open(fifo_path, 'w') as fifo:
        # blockiert, bis Pipe durch lesenden Prozess geöffnet wird
        fifo.write("Daten des schreibenden Prozesses.\n")

File: ipc/vorgabe/02/pipe_writer.cpp
#include <sys/stat.h>  // mkfifo
#include <fcntl.h>     // open
#include <unistd.h>    // write, close
#include <cstring>     // strlen

int main()
{
    // Named Pipe anlegen und öffnen
    const char *fifo_path = "/tmp/my_pipe";
    mkfifo(fifo_path, 0666);
    int fd = open(fifo_path, O_WRONLY); // Write Only

    // Nachricht in Named Pipe schreiben
    const char *message = "Daten des schreibenden Prozesses.\n";
    write(fd, message, strlen(message));

    close(fd);
    return 0;
}
